### PR TITLE
Task/company crud

### PR DESCRIPTION
**Pull request by marccolm**
Components for listing, updating, deleting and creating a company (Client model) based on previous work on contact model.

### CODE REVIEW

**Comment by ingJuanGro** [comment]
It's done

**Comment by ingJuanGro** [comment]
It's done

File: src/app/accounts/clients/client-form.component.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                </div>
+            </div>
+
+            <div class="col-md-6">
+                <!-- Name field -->
+                <label for="recipient-address" class="form-control-label">Dirección:</label>
+                <div class="input-group">
+                    <span class="input-group-addon"><i class="fa fa-map-marker"></i>
+                    </span>
+                    <input type="text" placeholder="Starbucks" class="form-control" id="recipient-address" minlength="3" maxlength="24" name="address" [(ngModel)]="client.address" #address="ngModel" required>
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.html]: Change placeholder in 'Address' for something more helpful

File: src/app/accounts/clients/client-form.component.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                </div>
+                <!-- Feedback address -->
+                <div *ngIf="address.errors && address.touched" class="alert alert-danger">
+                    <div [hidden]="!address.errors.required">
+                        El nombre es obligatorio.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.html]: This is not the address. Please change 'nombre' for 'dirección'

File: src/app/accounts/clients/client-form.component.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    </div>
+                    <div [hidden]="!address.errors.minlength">
+                        El nombre debe ser mayor o igual a 3 caracteres.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.html]: Same as above

File: src/app/accounts/clients/client-form.component.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                    </div>
+                    <div [hidden]="!address.errors.maxlength">
+                        El nombre debe ser menor a 25 caracteres.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.html]: Same as above

File: src/app/accounts/clients/client-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+* - Request actions in modals to the parent component.
+**/
+export class ClientFormComponent implements OnChanges {
+  // Receives the client selected by the user or the empty object to know if is called the update or create client form.
+  @Input() client: Client;
+  // Requests close of the current modal to parent component.
+  @Output() requestCloseModal: EventEmitter<string> = new EventEmitter();
+  // Requests to parent component the show of the danger modal to confirm if the client is permanent removed.
+  @Output() requestWarning: EventEmitter<string> = new EventEmitter();
+  // Event for parent to push the new Contact to the list.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.ts]: This comment is CopyPasted... please fix

File: src/app/accounts/clients/client-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  @Output() clientCreated: EventEmitter<Client> = new EventEmitter();
+  // Event for parent to update the currentContact.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.ts]: This comment is CopyPasted... please fix

File: src/app/accounts/clients/client-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  /**
+  * Update the Client object to avoid changes with TWDB.
+  * Params:
+  *   - object: A Client object.
+  *   - id: Id of client object.
+  **/
+  public cancelUpdatedClient(object: Client, id: number) {
+    let updatedClient = new Client(object);
+    this.clientUpdated.emit(updatedClient);
+  }
```
Comment (EduardoVaca) [inline src/app/accounts/clients/client-form.component.ts]: Get rid of this method, it is not needed. Put this.clientUpdated.emit(updatedClient); in the method calling this method.

File: src/app/accounts/clients/clients.component.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  * Params:
+  *   - url: The url where the service will comunicate to get the Client object.
+  * Returns:
+  *   - clientsList: Clients list with all the clients.
+  **/
+  public loadClientsList(url: string) {
+    this.httpService.getObject(url)
+                    .map((data: any) => data.json())
+                    .subscribe(clientsListJSON => {
+                      // Creates Iguala objects from JSON.
```
Comment (EduardoVaca) [inline src/app/accounts/clients/clients.component.ts]: This is not Iguala objects

Comment (JuanGro) [inline src/app/accounts/clients/clients.component.ts]: Done

File: src/app/accounts/clients/client-form.component.html (hunk 1 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.html]: Done

File: src/app/accounts/clients/client-form.component.html (hunk 2 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.html]: Done

File: src/app/accounts/clients/client-form.component.html (hunk 3 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.html]: Done

File: src/app/accounts/clients/client-form.component.html (hunk 4 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.html]: Done

File: src/app/accounts/clients/client-form.component.ts (hunk 5 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.ts]: Done

File: src/app/accounts/clients/client-form.component.ts (hunk 6 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.ts]: Done

File: src/app/accounts/clients/client-form.component.ts (hunk 7 above)
Comment (JuanGro) [inline src/app/accounts/clients/client-form.component.ts]: Done

**Pull request merged**